tests/by-util/test_ls.rs

```diff
@@ -3929,15 +3929,66 @@ fn test_ls_perm_io_errors() {
 }
 
 #[test]
-fn test_ls_dired_incompatible() {
+fn test_ls_dired_implies_long() {
     let scene = TestScenario::new(util_name!());
 
     scene
         .ucmd()
         .arg("--dired")
-        .fails()
-        .code_is(1)
-        .stderr_contains("--dired requires --format=long");
+        .succeeds()
+        .stdout_does_not_contain("//DIRED//")
+        .stdout_contains("  total 0")
+        .stdout_contains("//DIRED-OPTIONS// --quoting-style");
+}
+
+#[test]
+fn test_ls_dired_hyperlink() {
+    // we will have link but not the DIRED output
+    // note that the order matters
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.mkdir("dir");
+    at.touch("dir/a");
+    scene
+        .ucmd()
+        .arg("--dired")
+        .arg("--hyperlink")
+        .arg("-R")
+        .succeeds()
+        .stdout_contains("file://")
+        .stdout_does_not_contain("//DIRED//");
```

Comment:
While this test passes, the output is different from GNU `ls`:
```
$ ls --dired --hyperlink -R ../playground/ls_6144
../playground/ls_6144:
total 4
drwxr-xr-x 2 dho dho 4096 Jun 23 14:56 dir

../playground/ls_6144/dir:
total 0
-rw-r--r-- 1 dho dho 0 Jun 23 14:56 a

$ cargo run ls --dired --hyperlink -R ../playground/ls_6144
../playground/ls_6144:
dir

dir:
a
```
If `--dired` and `--hyperlink` are swapped, both uutils and GNU show the same output.

Reply:
one more time, I am impressed by your review skills, bravo :)

this is probably a small bug on the gnu bug.
hyperlink overrides only part of dired